saleor/core/middleware.py

```diff
@@ -28,25 +28,101 @@
 logger = logging.getLogger(__name__)
 
 
-def django_only_request_handler(get_response: Callable, handler: Callable):
+class RestrictedPathRequestHandler(MiddlewareMixin):
+    """Wraps a middleware to restrict it to only perform
+    when a given condition is True."""
+
+    # The methods to wrap with the path check
+    # Refer to https://docs.djangoproject.com/en/2.2/topics/http/middleware/
+    MIDDLEWARE_METHODS = (
```

Comment:
`process_request` doesn't have to be included here?

Reply:
This is actually dead code. The first idea was to dynamically wrap methods, which was why there was this list.

It was actually a bad idea for the reasons, that:
1. Too much magic
2. The methods don't all return `None`, one needs to call the chain, another needs the response returned.

Having no magic allows better flexibility and to introduce new methods if any other methods get added.